### PR TITLE
Move multi-var detection out of NL server

### PR DESCRIPTION
**Pull request by pradh**
With the intent of getting rid of the NL server, strip down its functionality to compute query embeddings and look it up in its index.  This will be followed by a real "Vector DB" support.

This is a ZERO functionality change PR.

TODO:  Due to this move there is one multi-var embeddings test that needs to be rewritten in the explore_test framework.  Since there is a concurrent PR (https://github.com/datacommonsorg/website/pull/4092) modifying those files, I'll make that test change in a follow up PR.

### CODE REVIEW

**Review by chejennifer — APPROVED** [review]
Nice! Only thing is I see some changes moving towards renaming sv -> vars but also not renaming it everywhere, wonder why the change and should we try to be consistent with var/sv?

File: nl_server/embeddings.py [hunk 1]
```diff
@@ -82,10 +72,9 @@ def __init__(self, embeddings_path: str, model: any) -> None:
   # Given a list of queries, searches the in-memory embeddings index
   # and returns a map of candidates keyed by input queries.
   #
-  def _search_embeddings(
-      self,
-      queries: List[str],
-      skip_topics: bool = False) -> Dict[str, vars.VarCandidates]:
+  def search_vars(self,
```
[the remark this answers is not in the document]
Comment (pradh) [reply nl_server/embeddings.py:75]: With multi-sv detection out of NL server, the endpoint is really a simple "vector search" endpoint?  Thus the renaming from detect to search.

**Comment by pradh** [comment]
> Nice! Only thing is I see some changes moving towards renaming sv -> vars but also not renaming it everywhere, wonder why the change and should we try to be consistent with var/sv?

Sorry for not explaining the rationale.  I was thinking of renaming things such that "vars" can mean => StatVar or Topic or Prop, whereas "sv" just means StatVar.

But wanted to limit the scope of the changes.  So, in this PR, renaming just the:
* The high-level variable detection entrypoint "detect_vars"
* The Lower level embeddings search endpoint "search_vars"

Wdyt?

**Comment by pradh** [comment]
> > Nice! Only thing is I see some changes moving towards renaming sv -> vars but also not renaming it everywhere, wonder why the change and should we try to be consistent with var/sv?
> 
> Sorry for not explaining the rationale. I was thinking of renaming things such that "vars" can mean => StatVar or Topic or Prop, whereas "sv" just means StatVar.
> 
> But wanted to limit the scope of the changes. So, in this PR, renaming just the:
> 
> * The high-level variable detection entrypoint "detect_vars"
> * The Lower level embeddings search endpoint "search_vars"
> 
> Wdyt?

I'll submit this, to follow up on test changes, we can discuss further name changes next week.

**Pull request merged**